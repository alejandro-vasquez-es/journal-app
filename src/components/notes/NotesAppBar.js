import React from 'react'

export const NotesAppBar = () => {
	return (
		<div className="notes__appbar">
			<span> 16 de febrero del 2022 </span>

			<div className='notes__btn-container'>
				<button className='btn'>
					Picture
				</button>
				<button className='btn'>
					Save
				</button>
			</div>
		</div>
	)
}
