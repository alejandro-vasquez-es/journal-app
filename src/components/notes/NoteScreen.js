import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
	return (
		<div className="notes__main-content">

			<NotesAppBar />

			<div className='notes__content'>

				<input
					type="text"
					placeholder="jijijajaja"
					className="notes__title-input"
				/>

				<textarea
					placeholder='what happened today'
					className='notes__textarea'
				>

				</textarea>

				<div className='notes__image'>
					<img
						alt='landcape'
						src='https://images.pexels.com/photos/592077/pexels-photo-592077.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500'
					/>
				</div>

			</div>

		</div>
	)
}
