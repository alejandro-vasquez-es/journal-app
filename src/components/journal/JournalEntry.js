import React from 'react';
import moment from 'moment'
import { useDispatch } from 'react-redux';
import { activeNote } from '../../actions/notes';

export const JournalEntry = ({ id, date, title, body, url }) => {

	const dispatch = useDispatch();
	const noteDate = moment(date);

	const goToActiveNote = (duration) => {
		const targetPosition = Math.max(document.documentElement.clientHeight || 0, window.innerHeight || 0);
		var startPosition = window.pageYOffset;
		var distance = targetPosition - startPosition;
		var startTime = null;

		function animation(currentTime) {
			if (startTime === null) startTime = currentTime;
			var timeElapsed = currentTime - startTime;
			var run = easeOutExpo(timeElapsed, startPosition, distance, duration);
			window.scroll(0, run);
			if (timeElapsed < duration) requestAnimationFrame(animation);
		}

		function easeOutExpo(t, b, c, d) {
			return c * (-Math.pow(2, -10 * t / d) + 1) + b - 16;
		}

		requestAnimationFrame(animation);
	}

	const handleEntryClick = () => {
		dispatch(activeNote(id, { date, title, body, url }));
		goToActiveNote(500);
	}

	return (
		<div
			className='journal__entry  animate__animated animate__slideInLeft animate__faster'
			onClick={handleEntryClick}
		>

			{
				url &&
				<div
					className='journal__entry-picture'
					style={{
						backgroundSize: 'cover',
						backgroundImage: `url(${url})`,
						backgroundPosition: 'center'
					}}
				>

				</div>
			}

			<div className='journal__entry-body'>
				<p className='journal__entry-title'>
					{title ? title : 'Untitled'}
				</p>
				<p className='journal__entry-content'>
					{body}
				</p>
			</div>

			<div className='journal__entry-date-box'>
				<p>{noteDate.format('dddd')}</p>
				<span>{noteDate.format('DD')}</span>
			</div>

		</div>
	)
}
