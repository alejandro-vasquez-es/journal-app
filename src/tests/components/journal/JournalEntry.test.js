import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { Sidebar } from '../../../components/journal/Sidebar';
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();

const note = {
	id: 18,
	date: new Date(),
	title: 'TITLE',
	body: 'BODY',
	url: 'https://someplace.com/photo.png'
}

const wrapper = mount(
	<Provider store={store}>
		<JournalEntry {...note} />
	</Provider>
);

describe('tests in <JournalEntry/>', () => {

	test('should display propertly', () => {

		expect(wrapper).toMatchSnapshot();

	});

	test('should activate the note', () => {

		wrapper.find('.journal__entry').simulate('click');

		expect(store.dispatch).toHaveBeenCalledWith(activeNote(note.id, { ...note }));

	})

})