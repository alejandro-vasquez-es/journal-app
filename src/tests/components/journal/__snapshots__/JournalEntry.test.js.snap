// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <JournalEntry/> should display propertly 1`] = `
<div
  className="journal__entry  animate__animated animate__slideInLeft animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://someplace.com/photo.png)",
        "backgroundPosition": "center",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      TITLE
    </p>
    <p
      className="journal__entry-content"
    >
      BODY
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <p>
      Friday
    </p>
    <span>
      25
    </span>
  </div>
</div>
`;
